components/gitpod-cli/cmd/rebuild.go

```
@@ -0,0 +1,234 @@
+		if err != nil {
+			utils.LogError(ctx, err, "Could not fetch the workspace info", client)
+			return
+		}
+
+		tmpDir, err := os.MkdirTemp("", "gp-rebuild-*")
+		if err != nil {
+			log.Fatal("Could not create temporary directory")
+			return
+		}
+		defer os.RemoveAll(tmpDir)
+
+		event := utils.TrackEvent(ctx, client, &utils.TrackCommandUsageParams{
+			Command: cmd.Name(),
+		})
+		defer event.Send(ctx)
+
+		gitpodConfig, err := utils.ParseGitpodConfig(wsInfo.CheckoutLocation)
+		if err != nil {
+			fmt.Println("The .gitpod.yml file cannot be parsed: please check the file and try again")
+			fmt.Println("")
+			fmt.Println("For help check out the reference page:")
+			fmt.Println("https://www.gitpod.io/docs/references/gitpod-yml#gitpodyml")
+			event.Set("ErrorCode", utils.RebuildErrorCode_MalformedGitpodYaml)
+			return
+		}
+
+		if gitpodConfig == nil {
+			fmt.Println("To test the image build, you need to configure your project with a .gitpod.yml file")
+			fmt.Println("")
+			fmt.Println("For a quick start, try running:\n$ gp init -i")
+			fmt.Println("")
+			fmt.Println("Alternatively, check out the following docs for getting started configuring your project")
+			fmt.Println("https://www.gitpod.io/docs/configure#configure-gitpod")
+			event.Set("ErrorCode", utils.RebuildErrorCode_MissingGitpodYaml)
+			return
+		}
+
+		var baseimage string
+		switch img := gitpodConfig.Image.(type) {
+		case nil:
+			baseimage = ""
+		case string:
+			baseimage = "FROM " + img
+		case map[interface{}]interface{}:
+			dockerfilePath := filepath.Join(wsInfo.CheckoutLocation, img["file"].(string))
+
+			if _, err := os.Stat(dockerfilePath); os.IsNotExist(err) {
+				fmt.Println("Your .gitpod.yml points to a Dockerfile that doesn't exist: " + dockerfilePath)
+				event.Set("ErrorCode", utils.RebuildErrorCode_DockerfileNotFound).Send(ctx)
+				return
+			}
+			dockerfile, err := os.ReadFile(dockerfilePath)
```

err is swallowed, we should give a root cause to the user